src/deadline/client/dataclasses/submitter_info.py

```diff
@@ -0,0 +1,54 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+"""
+Dataclass for holding submitter environment information.
+"""
+
+from __future__ import annotations
+
+__all__ = ["SubmitterInfo"]
+
+from dataclasses import dataclass
+from typing import Any, Optional
+
+
+@dataclass
+class SubmitterInfo:
+    """
+    Container for submitter environment information.
+
+    This dataclass holds metadata about the application submitting jobs to AWS Deadline Cloud.
+    It's used by the _AboutDialog GUI to display version information.
+
+    Attributes:
+        submitter_name: Short name of the submitter (required, e.g., "Blender", "CLI")
+        submitter_package_name: Name of the submitter package using the library (optional, e.g., "deadline-cloud-for-blender")
+        submitter_package_version: Version of the submitter package (optional)
+        host_application_name: Name of the host application (optional, e.g., "Maya", "Blender")
+        host_application_version: Version of the host application (optional)
+        additional_info: Optional dictionary for arbitrary nested data that integrations can use
+                        to pass additional metadata (optional). Supports nested structures.
```

Comment:
nit: can we add these as docstrings to the attributes below?

This improves the dev experience in supported IDEs that will show the help text. For example in VSCode when hovering over the attribute.

An example docstring for an attribute is:

```py
    submitter_name: str
    """Short name of the submitter (required, e.g., "Blender", "CLI")"""
```